cmd/serverless/main_test.go

```diff
@@ -9,83 +9,47 @@ package main
 
 import (
 	"os"
-	"path"
-	"strings"
+	"syscall"
 	"testing"
+	"time"
 
-	"github.com/stretchr/testify/assert"
-
-	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/serverless/daemon"
 )
 
-func setupTest() {
-	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
-	config.InitConfig(config.Datadog)
-	os.Setenv("AWS_LAMBDA_FUNCTION_NAME", "TestFunction")
-}
-
-func TestProxyNotLoaded(t *testing.T) {
-	setupTest()
-
-	proxyHTTP := "a:1"
-	proxyHTTPS := "a:2"
-	t.Setenv("DD_PROXY_HTTP", proxyHTTP)
-	t.Setenv("DD_PROXY_HTTPS", proxyHTTPS)
-	proxyHTTPConfig := config.Datadog.GetString("proxy.http")
-	proxyHTTPSConfig := config.Datadog.GetString("proxy.https")
-	assert.Equal(t, 0, len(proxyHTTPConfig))
-	assert.Equal(t, 0, len(proxyHTTPSConfig))
-}
-
-func TestProxyLoadedFromEnvVars(t *testing.T) {
-	setupTest()
+func TestDaemonStopOnSIGINT(t *testing.T) {
+	stopCh := make(chan struct{})
+	signalCh := make(chan os.Signal, 1)
 
-	proxyHTTP := "b:1"
-	proxyHTTPS := "b:2"
-	t.Setenv("DD_PROXY_HTTP", proxyHTTP)
-	t.Setenv("DD_PROXY_HTTPS", proxyHTTPS)
+	serverlessDaemon := daemon.StartDaemon("http://localhost:8124")
+	go handleTerminationSignals(serverlessDaemon, stopCh, signalCh)
 
-	config.LoadWithoutSecret()
-	proxyHTTPConfig := config.Datadog.GetString("proxy.http")
-	proxyHTTPSConfig := config.Datadog.GetString("proxy.https")
+	signalCh <- syscall.SIGINT
 
-	assert.Equal(t, proxyHTTP, proxyHTTPConfig)
-	assert.Equal(t, proxyHTTPS, proxyHTTPSConfig)
+	// Use t.Run with a timeout to allow the goroutine to finish
+	t.Run("WaitForStop", func(t *testing.T) {
```

Comment:
I think that's because the goroutines are stopped before they have a chance to finish and assert what is being tested. By having this `t.Run` function we allow for 1 second to receive the expected signal in `stopCh` and fail if 1 second passes before receiving the signal.

If you comment out this line the test should fail:
https://github.com/DataDog/datadog-agent/blob/b135597e2144573293058fed625291139f72ca2a/cmd/serverless/main_test.go#L26